### ISSUE TITLE
Website CSP can block AdGuard scripts (Firefox)

### ISSUE THREAD

**Issue by HanabishiRecca** [AdguardTeam/AdguardBrowserExtension#1733]
- **AdGuard version:** 3.5.23
- **Browser and version:** Firefox 83.0

Some websites using strict CSP rules can block AdGuard scripts injection.
E.g. here at `github.com` AdGuard can't make it's own scripts to work.
![image](https://user-images.githubusercontent.com/13597663/101951981-bdbc1100-3c19-11eb-8b61-c138c413cb77.png)
So all JS-related filters doesn't work with the such websites.

I've tried to override this behavior with a rule like this:
`github.com^$csp=default-src script-src 'unsafe-inline' 'unsafe-eval' *`
But it doesn't work for some reason.

This is Firefox-only issue. Seems like Chrome allows extensions to inject scripts regardless of CSP.

**Comment by Chinaski1** [comment]
Hello!

It looks like there should be no CSP check for extension resources.
Apparently, this is a bug in the Firefox browser.

**Comment by isaak654** [comment]
I constantly see a red Content Security Policy warning in Firefox Console during the normal navigation on Github.com too.

![firefox_console](https://user-images.githubusercontent.com/12372772/105384729-e240db00-5c12-11eb-8209-7a6b906f2935.png)

I'm using Firefox 84.0.2 x64. With the same browser I can't make more than 5 edits to a comment because my changes are not correctly retained between an edit and another. Currently I'm waiting official Github confirmation about it.

I've already tried without success:
1) to disable "Filter update interval" option in Adguard
2) to include github.com in the Adguard whitelist

**Comment by HanabishiRecca** [comment]
> With the same browser I can't make more than 5 edits to a comment because my changes are not correctly retained between an edit and another.

I don't think this is related. CSP errors means exactly the opposite: Adguard scripts don't work at this website. So there is literally nothing to interfere with.

**Comment by XX-J** [comment]
А я то думал, - почему scriptlet'ы на некоторых сайтах не работают, а тут вон оно что...
 Не работают scriptlet'ы на GitHub'е, Яндексе и других сайтах, где они ОЧЕНЬ нужны.
<br>

 Система | Фильтры | Антитрекинг
---------- | ---------- | ----------
 AdGuard v3.6.13 | Базовый AdGuard &nbsp; &nbsp; &nbsp; &nbsp; Русский | Стандарт
 FireFox v78.13 ESR | Счётчиков и аналитики &nbsp; &nbsp; &nbsp; &nbsp; Отслеживания по URL | +Блокировка WebRTC
 Win7SP1 | Виджетов социальных сетей &nbsp; &nbsp; &nbsp; &nbsp; Раздражителей | +Удаление параметров отслеживания

**Comment by XX-J** [comment]
Workaround this bug in FireFox is put `security.csp.enable = false` in about:config.

 Странно, но GreaseMonkey как-то обходит эту проблему, а AdGuard и TamperMonkey - нет. Почему и, самое главное, Как?

**Comment by XX-J** [comment]
> Точно? Тогда нужно покопаться в его исходнике просто.

 Точно! Использую версию 4.11 на FireFox v78.13 ESR, только он в iframe'ах скрипты не запускает. А TamperMonkey работает в iframe'ах, но не работает при строгом CSP.

 А в коде я настолько сильно не разбираюсь - надо более опытного кодера.

**Comment by XX-J** [comment]
А ОбезьянаВмасле работает по такому принципу - удаляет из заголовка обрабатываемых страниц параметр `content-security-policy`?

**Comment by XX-J** [comment]
Я считаю **такой** способ решения проблемы для **AdGuard'a** сопоставим с отключением CSP, т.к. скрипты/скриптлеты необходимы/работают на очень многих страницах. Хотя, в крайнем случае, я бы согласился на опцию автовырезания CSP из заголовков тех доменов/страниц, которые:  1. Открываются в FireFox'e;  2. Для них есть правила со скриптами/скриптлетами.

 Но, если в МаслянойОбезьяне применён другой, более хитрый метод обхода, то, думаю, он был бы предпочтительнее...

**Comment by XX-J** [comment]
> ...на опцию **автовырезания CSP из заголовков**...

 Тем самым **модифицируя** его (т.е. заголовок).

**Comment by FireFlashie** [comment]
@maximtop, there does seem to workaround this, at least for most cases.
https://github.com/uBlockOrigin/uBlock-issues/issues/235#issuecomment-1344313926

**Comment by maximtop** [comment]
@FireFlashie thank you

**Comment by krystian3w** [comment]
#2276
